java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -189,12 +192,34 @@ public boolean useProlepticGregorian() {
     public boolean fileUsedProlepticGregorian() {
       return fileUsedProlepticGregorian;
     }
+
+    @Override
+    public TypeReader.ReaderCategory getReaderCategory(int columnId) {
+      TypeReader.ReaderCategory result;
+      if (getColumnFilterCallback() == null || getColumnFilterIds().contains(columnId)) {
```

Comment:
Not sure why consider getColumnFilterCallback null case here? 
When there is not filter set we consider all ReaderCategories to be FILTER_PARENT or FILTER_CHILD?

Reply:
Good catch. I missed cleaning up this condition, I already inverted in the previous commit that in the absence of filters we treat every column as NON_FILTER. Am removing `getColumnFilterCallback() == null` condition.